packages/block-library/src/navigation/edit.js

```diff
@@ -134,7 +155,7 @@ function Navigation( {
 		selectBlock( clientId );
 	}
 
-	const hasPages = hasResolvedPages && pages && pages.length;
+	const hasPages = pages && pages.length;
```

Comment:
I think best to cast this to a boolean so that there's no surprises down the road when someone uses `hasPages` expecting a boolean but gets e.g. a number.

```suggestion
	const hasPages = !! pages && !! pages.length;
```